core/src/main/java/org/apache/calcite/sql/SqlDialect.java

```diff
@@ -797,6 +797,11 @@ public boolean supportBooleanCaseWhen() {
     return true;
   }
 
+  /** Returns whether this dialect supports generate 'SELECT *' when duplicate field names. */
+  public boolean supportGenerateSelectStar() {
```

Comment:
The comment for this method is somewhat ambiguous. From an implementation perspective, it is not solely controlled by this method.

Reply:
I agree with your comment. Indeed, I didn't come up with a good method name and code comments here. I will think about it again.

Reply:
I suggest passing in the RelNode as a parameter, and then different Dialects can make more customized decisions based on their specific circumstances regarding whether to generate star.

Reply:
Great suggestion, this makes the code logic clearer. I will revise according to this suggestion later.